substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -539,10 +495,19 @@ where
 			return Box::new(std::iter::empty());
 		};
 
+		let at_hn = HashAndNumber { hash: at, number: at_number };
+		let last_finalized = self.enactment_state.lock().recent_finalized_block();
+		// Return an empty ready txs set as well if we can not resolve the recent finalized block
+		// number.
+		let Some(last_finalized_number) = self.api.resolve_block_number(last_finalized).ok() else {
+			return Box::new(std::iter::empty());
```

Comment:
I think we misunderstood each other, sorry for not being clear enough.
In this case we can also return recent best block - no reason to provide empty set.

Reply:
Do you think it is worth doing the same in case we can not relsolve `at` block numer too?

Reply:
Changed it so: [15afbb3](https://github.com/paritytech/polkadot-sdk/pull/8533/commits/15afbb349c9a4d6932c819e878ad9933febdc066). Didn't make a change for the question above.

Reply:
> Do you think it is worth doing the same in case we can not relsolve `at` block numer too?

Yes. If anything goes the most recent view should be returned.
But I think it is already handled - if we can't resolve `at` header - `find_view_descendent_upto_number` will return `None`. So should be good.

Reply:
Resolving in favour of https://github.com/paritytech/polkadot-sdk/pull/8533#discussion_r2095889942.